fvm/transactionInvoker.go

```diff
@@ -69,28 +74,33 @@ func (i *TransactionInvoker) Process(
 		if sth.NumNestedTransactions() > 1 {
 			err := sth.RestartNestedTransaction(nestedTxnId)
 			if err != nil {
-				processErr = fmt.Errorf(
-					"cannot restart nested transaction: %w",
-					err,
+				processErr = errors.FirstOrFailure(
+					logger,
+					processErr,
+					fmt.Errorf(
+						"cannot restart nested transaction: %w",
+						err,
+					),
 				)
 			}
 
 			msg := "transaction has unexpected nested transactions"
-			i.logger.Error().
-				Str("txHash", txIDStr).
-				Uint64("blockHeight", blockHeight).
+			logger.Error().
 				Msg(msg)
 
 			proc.Err = errors.NewFVMInternalErrorf(msg)
 			proc.Logs = make([]string, 0)
 			proc.Events = make([]flow.Event, 0)
 			proc.ServiceEvents = make([]flow.Event, 0)
-
 		}
 
 		err := sth.Commit(nestedTxnId)
 		if err != nil {
-			processErr = fmt.Errorf("transaction invocation failed when merging state: %w", err)
+			processErr = errors.FirstOrFailure(
```

Comment:
Same question here, do we still want to commit, if processErr is a failure already?

If we don't run commit when processErr is failure, then we won't have the `err` to handle.